Auto Set Resolution on Scene; Add UI_Scale

KerballOne opened the pull request:
It can now automatically switch resolution and UI scale based on scene.  All it needs is a config file with the scene name (e.g. 'EDITOR', 'FLIGHT', etc.).  Each setting config now includes a "scale" attribute in addition to name, height, width and fullscreen.
The UI Window updates after each action so saving and/or apply new presets is easier.

KerballOne committed:
New Config Files

Added config files to match SCENE names
Added scale attribute

KerballOne committed:
Per Scene, UI Scale

 1)  Include UI Scale alongside Resolution.

2)  Remember resolution and scale per scene and automatically switch when switching scenes

KerballOne committed:
GUI Window updates with current Res/scale

Text input fields of the GUI Toolbar Window updates when changing to new scene, Clicking Set, or Clicking preset

KerballOne committed:
New default config files

KerballOne committed:
Delete GameData/AnyRes/PluginData directory

KerballOne committed:
New default config files

KerballOne committed:
Update AnyRes.cs

Typo line 41

linuxgurugamer commented:
I had to fix a performance issue, which is now causing conficts, can you resolve them and let mw know when it's ready?

KerballOne committed:
Added UI Scale

It can now automatically switch resolution and UI scale based on scene. All it needs is a config file with the scene name (e.g. 'EDITOR', 'FLIGHT', etc.). Each setting config now includes a "scale" attribute in addition to name, height, width and fullscreen.
The UI Window updates after each action so saving and/or apply new presets is easier.

KerballOne committed:
Merge branch 'master' into master

KerballOne commented:
Resolved conflicts after remodding the code.  I think/hope I did it right.  I am new to C#, VS and even git, so possible I am doing things wrong or inefficiently.

linuxgurugamer commented:
I've merged it into a branch.  It did take a minute to figure out what was going on



Specifically, it seems that it ignored the previous setting entirely.  I installed it in an existing game, which had been set to 1920x1080 windowed, and when it started, it automatically went to full screen.

I see you also added an auto-ui scale, this was also very confusing.

Essentially, you are making assumptions about what the user would want, which is always very iffy.

I recommend the following changes/additions:

1. Write a bit more code that first checks to see if those files are there, if not, then create them using the current resolution settings and UI scale of the game.  That would be less intrusive
2. Add a slider to the config window to set the UI scale.  This will probably require the window to be a bit bigger
